src/LieTheory/WeylGroup.jl

```diff
@@ -516,6 +516,42 @@ function word(x::WeylGroupElem)
   return x.word
 end
 
+@doc raw"""
+    reflection(beta::RootSpaceElem) -> WeylGroupElem
+
+Return the Weyl group element corresponding to the reflection at the hyperplane orthogonal to the root `beta`.
+"""
+function reflection(beta::RootSpaceElem)
+  R = root_system(beta)
+  W = weyl_group(R)
+  rk = number_of_simple_roots(R)
+
+  b, index_of_beta = is_positive_root_with_index(beta)
+  if !b
+    b, index_of_beta = is_negative_root_with_index(beta)
+  end
+  @req b "Not a root"
```

Comment:
Please add a comment why `is_positive_root_with_index` is not appropriate here.

Reply:
Also we should improve the doc string of `is_negative_root_with_index`, since the index returned is the index of the positive root, while `is_positive_root_with_index` returns the index of the negative root.